secboot/secboot_hooks.go

```diff
@@ -174,9 +166,21 @@ func unlockVolumeUsingSealedKeyFDERevealKeyV2(sealedEncryptionKeyFile, sourceDev
 		return res, xerrors.Errorf(fmt, err)
 	}
 
+	model, err := opts.WhichModel()
+	if err != nil {
+		return res, fmt.Errorf("cannot retrieve which model to unlock for: %v", err)
+	}
+
 	// the output of fde-reveal-key is the unsealed key
 	options := activateVolOpts(opts.AllowRecoveryKey)
-	modChecker, err := sbActivateVolumeWithKeyData(mapperName, sourceDevice, keyData, options)
+	options.Model = model
```

Comment:
nitpick: maybe it would make sense if `activateVolOpts()` accepts the model as input so it creates `ActivateVolumeOptions` with it already instead of assigning a line later. As this is done in the 2 cases where this function is called.

Reply:
Well, since we will remove this one and call SetModel instead, I prefer not to change it.

Reply:
Maybe this is not completely true. For the old keys will still need this to be set. But we will possibly we will only add it when we have old key files and not when we have key slot token data. Then we can set it to nil to make it force to fail if it finds old data. I will add a FIXME here.

Reply:
Also both unlock for FDE hook and tpm will be joined in the same function at some point.